url4e79c36700_JacenLakiir_Tivoo_tgz-pJ8-MainJ8/src/tivoo/output/HTMLBuilder.java

```diff
@@ -0,0 +1,382 @@
+                sortedEvents.put(eventDay, new ArrayList<Event>());
+            sortedEvents.get(eventDay).add(currentEvent);
+        }
+        return sortedEvents;
+    }
+    
+    protected Map<String, List<Event>> sortByDate (List<Event> eventList)
+    {
+        Map<String, List<Event>> sortedEvents = new TreeMap<String, List<Event>>();
+        for (Event currentEvent : eventList)
+        {
+            String eventDay = formatShortDate(currentEvent);
+            if (!sortedEvents.containsKey(eventDay))
+                sortedEvents.put(eventDay, new ArrayList<Event>());
+            sortedEvents.get(eventDay).add(currentEvent);
+        }
+        return sortedEvents;
+    }
+    
+    protected String getTitle ()
+    {
+        return TITLE;
+    }
+    
+    protected String getUniqueCSS ()
+    {
+        return UNIQUE_CSS;
+    }
+    
+    private static List<String> initializeDaysList ()
+    {
+        String[] days =
+            new String[] {
+                    "Sunday",
+                    "Monday",
+                    "Tuesday",
+                    "Wednesday",
+                    "Thursday",
+                    "Friday",
+                    "Saturday" };
+        List<String> dayList = new ArrayList<String>();
+        for (String d : days)
+            dayList.add(d);
+        return dayList;
+    }
+    
+    private static List<String> initializeMonthsList ()
+    {
+        String[] months =
+            new String[] {
+                    "January",
+                    "February",
+                    "March",
+                    "April",
+                    "May",
+                    "June",
+                    "July",
+                    "August",
+                    "September",
+                    "October",
+                    "November",
+                    "December" };
+        List<String> monthList = new ArrayList<String>();
+        for (String m : months)
+            monthList.add(m);
+        return monthList;
+    }
+
+    private String determineDetailPageDirectory (String summaryPageFileName)
+    {
+        StringBuilder detailPageDirectory = new StringBuilder();
+        if (summaryPageFileName.contains("/"))
+        {
+            int lastFolder = summaryPageFileName.lastIndexOf("/");
+            detailPageDirectory.append(summaryPageFileName.substring(0, lastFolder + 1));
+            detailPageDirectory.append(DETAIL_PAGE_FOLDER);
+        }
+        else
+        {
+            detailPageDirectory.append(DETAIL_PAGE_FOLDER);
+        }
+        return detailPageDirectory.toString();
+    }
+    
+    private void buildSummaryPage (List<Event> eventList) throws IOException
+    {
+        if (mySummaryPageFileName.contains("/"))
+        {
+            File outputDirectory = new File(myDetailPageDirectory);
+            outputDirectory.mkdirs();
+        }
+        File summaryPage = new File(mySummaryPageFileName);
+        summaryPage.createNewFile();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

`File.createNewFile` returns `false` if the named file already exists and an empty file is not created. Check the value returned by `createNewFile()`. Handle when the method returns `false`. Otherwise, you might silently overwrite an existing file which could lead to data loss. See the following diffs on GitHub that make this code change: [diff1](https://github.com/lingochamp/FileDownloader/commit/65069c18a4b1b4cfa758173ef49d69c6897fc77a#diff-2b6ad96df9bf9828c22f961e9e9a5b933ab2df7b2fb0f12d62c044309e1e2842L331), [diff2](https://github.com/m-szalik/javamail/commit/62b1cb45e32a998efbe6cfa09b06752869f590ae#diff-440ef5498fc2c802274cf136e7bd7d682bbc10c5976afb5af718984e24deced4L45)

Learn more: [File.createNewFile in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#createNewFile--)